### PR TITLE
Improve task termination

### PR DESCRIPTION
**Pull request by jherbel**
In addition to attempting to kill the process via its PID, we create a
flag file. Before starting the actual test run, the task checks if the
flag file still exists. If not, the task terminates.

This provides us with an additional mechanism for shutting down the task
in its early stage when the PID might not yet have been written to file.

### CODE REVIEW

**Review by SoloJacobs — COMMENTED** [review]
I'm unsure how this mechanism works. Where is the checking for the flag file done? (I spent 10min looking, but still unclear, sorry!)

File: v2/robotmk/src/bin/scheduler/sessions/schtasks.rs [hunk 1]
```diff
@@ -161,6 +163,14 @@ where
     ))
 }
 
+fn start_task(task_name: &str, path_run_flag: &Utf8Path) -> Result<()> {
+    debug!("Starting task {task_name}");
+    write(path_run_flag, "").context(format!("Failed to create run flag file {path_run_flag}"))?;
```
Comment (SoloJacobs) [inline v2/robotmk/src/bin/scheduler/sessions/schtasks.rs:168]: Does it really make sense to mix this with the other errors?

**Review by SoloJacobs — APPROVED** [review]
Hmm, I don't think this is the correct approach, but it's simple and works... so oti.

File: v2/robotmk/src/bin/scheduler/sessions/schtasks.rs (hunk 1 above)
Comment (jherbel) [reply v2/robotmk/src/bin/scheduler/sessions/schtasks.rs:168]: Not sure, but let's leave it like this for now. Hopefully we anyway find a better approach in the future.

**Comment by jherbel** [comment]
> I'm unsure how this mechanism works. Where is the checking for the flag file done? (I spent 10min looking, but still unclear, sorry!)

The check happens in this line:
`format!("if not exist {} exit /b 1", paths.run_flag)`

This is one of the lines we write to batch file that is executed by the task. This line is placed after the line which writes the PID to file and before the line which executes the test run. When we want to cancel the task, we first delete the flag file and then we additionally try to retrieve the PID and terminate the process. In case the latter doesn't work, the task should still be in its initial phase (before starting the test run) and the removal of the file should make it terminate.

I fully agree that this approach is ugly and far from optimal, but I am out of ideas. I tested it and it seemed to to the job.

**Pull request merged**